From 1ae1a502cc6020a66be6f0c78927e444cb4d7111 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Sat, 10 Nov 2018 17:56:27 +0100
Subject: [PATCH 13/14] Revert screenshot changes to exclude black cutout

* The reverted code sets screenshot area to display size
  -> It breaks cropped screenshots because new size doesn't match
     requested width and height ( as per cropped definition )
  -> It breaks devices with custom hw rotation because
     width and height are swapped

Revert "Get screencap working correctly in landscape mode"

This reverts commit 9ad73bc2967d6d97d359cbad16996fcd7b83d30f.

Revert "Get screenshot to exclude black cutout"

This reverts commit f4e02874b260f85acb7eba2d61fe9af37e290c0f.

Change-Id: Id3def3adfb95e8c5e696bf86ed6dfc4f6bcaba87
---
 services/surfaceflinger/SurfaceFlinger.cpp | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 814f04426..7055eae69 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -4838,16 +4838,6 @@ status_t SurfaceFlinger::captureScreen(const sp<IBinder>& display, sp<GraphicBuf
     const sp<const DisplayDevice> device(getDisplayDeviceLocked(display));
     if (CC_UNLIKELY(device == 0)) return BAD_VALUE;
 
-    const Rect& dispScissor = device->getScissor();
-    if (!dispScissor.isEmpty()) {
-        sourceCrop.set(dispScissor);
-        // adb shell screencap will default reqWidth and reqHeight to zeros.
-        if (reqWidth == 0 || reqHeight == 0) {
-            reqWidth = uint32_t(device->getViewport().width());
-            reqHeight = uint32_t(device->getViewport().height());
-        }
-    }
-
     DisplayRenderArea renderArea(device, sourceCrop, reqHeight, reqWidth, rotation);
 
     auto traverseLayers = std::bind(std::mem_fn(&SurfaceFlinger::traverseLayersInDisplay), this,
-- 
2.17.1

